utils/avro-components-utils/src/main/scala/pl/touk/nussknacker/engine/avro/schemaregistry/confluent/formatter/UniversalMessageReader.scala

```diff
@@ -0,0 +1,31 @@
+package pl.touk.nussknacker.engine.avro.schemaregistry.confluent.formatter
+
+import io.circe.Json
+import io.confluent.kafka.schemaregistry.ParsedSchema
+import io.confluent.kafka.schemaregistry.avro.AvroSchema
+import io.confluent.kafka.schemaregistry.client.SchemaRegistryClient
+
+import java.nio.charset.StandardCharsets
+
+/**
+  * @param schemaRegistryClient schema registry client
+  */
+private[confluent] class UniversalMessageReader(schemaRegistryClient: SchemaRegistryClient) {
+
+  private val avroMessageReader = new ConfluentAvroMessageReader(schemaRegistryClient)
+
+  def readJson(jsonObj: Json, parsedSchema: ParsedSchema, subject: String): Array[Byte] = {
+    parsedSchema match {
+      case schema: AvroSchema => avroMessageReader.readJson(jsonObj, schema.rawSchema(), subject)
+      //todo: change JsonSchema
+      case s: ParsedSchema if s.schemaType() == "JSON" =>
```

Comment:
could we extract all such cases to one place? 
and have one method like
```
(ParsedSchema, some config?) => {
    asJson(obj, subject): Json
    readJson(json, subject): Array[Byte]
    type: TypingResult
    deserialize(...)...
}
```
so we know what is needed to introduce new schema type?

Reply:
Agree, but I'll do such refactor in one of the next PR's

Reply:
:+1: